tests/x11/gnucash.pm

```diff
@@ -34,9 +34,9 @@ sub run {
         send_key 'alt-tab' if check_var('DESKTOP', 'lxde');
         wait_screen_change { send_key 'alt-c' };
     }
-    send_key 'ctrl-q';    # Exit
-    assert_screen [qw(gnucash-save-changes generic-desktop)];
-    wait_screen_change { send_key 'alt-w' } if match_has_tag 'gnucash-save-changes';
+    # sometimes send_key for closing windows not working
+    send_key_until_needlematch 'gnucash-save-changes', "ctrl-q", 5, 5;
+    send_key 'alt-w';
```

Comment:
Is there a reason why you changed the line
```perl
wait_screen_change { send_key 'alt-w' } if match_has_tag 'gnucash-save-changes';
```
to
```perl
send_key 'alt-w';
```
?

I think, the old way does a more strict check.

Reply:
the problem is send_key 'ctrl+q' doesn't work sometimes. So repeat it 5 times, it should work.

Afterwards simply send_key 'alt-w' works then because gnacash-save-changes got matched.
In this case there is no reason to check the same needle twice.